daisybell/helpers/chat.py

```diff
@@ -0,0 +1,40 @@
+import torch
+from transformers import Pipeline, StoppingCriteria, StoppingCriteriaList
+
+
+class StopOnTokens(StoppingCriteria):
+    def __call__(
+        self, input_ids: torch.LongTensor, scores: torch.FloatTensor, **kwargs
+    ) -> bool:
+        stop_ids = [50278, 50279, 50277, 1, 0]
```

Comment:
What do these IDs signify and how were they chosen?

Reply:
These stop tokens are unique IDs that the language generator uses to determine when to stop generating new text. Whenever the generator encounters a token with one of these IDs, it stops generating further text. However, if it doesn't encounter any of these stop tokens, it will continue generating new tokens until it reaches the specified token limit, which can be configured.

These stop tokens are emitted probabilistically from the language model, and there are a total of five possible stop tokens. All this will be clarified in the code.